packages/lib/src/utils/ABIs.ts

```diff
@@ -20,7 +20,7 @@ interface FunctionInfo {
 export function buildCallData(contractClass: ContractFactory, methodName: string, args: any[]): CalldataInfo {
   const method = getABIFunction(contractClass, methodName, args);
   const argTypes = method.inputs.map((input) => input.type);
-  const callData = encodeCall(method.name, argTypes, args);
+  const callData = encodeCall(method.name, argTypes, parseCallValues(argTypes, args));
```

Comment:
I'm not sure if this is the best place to add `parseCallValues`. This function gets called from lib/App and lib/SimpleProject, which can be called directly from the js interface, where the user specifies actual values and we do not want to go through the parsing process. I'd consider moving this all the way up to the CLI, potentially cli/scripts/create and cli/scripts/update.

Reply:
I see what you mean, but we have the type data at ABI.ts via analysis of the ABI for the function being called: https://github.com/zeppelinos/zos/blob/master/packages/lib/src/utils/ABIs.ts#L21

In the CLI's scripts this information is not yet available.

Reply:
I see, but we do need a way to keep a consistent interface when the user interacts programatically. I'm not comfortable in having different behaviours when initializing a contract via SimpleProject versus via web3 or truffle.

Can we make the ABI types available higher up in the chain? Or add an optional flag `parseInitArgs` that controls whether they are parsed or not, and is set to true only from the CLI?